collectors/perf.cpp

```diff
@@ -79,14 +74,33 @@ static inline uint64_t makeup_booker_ci_config(int nodetype, int eventid, int by
     return config;
 }
 
-PerfCollector::PerfCollector(const Json::Value& config, const std::string& name) : Collector(config, name)
+PerfCollector::PerfCollector(const Json::Value& config, const std::string& name, bool enablePerapiPerf) : Collector(config, name), mEnablePerapiPerf(enablePerapiPerf)
 {
-    struct event leader = {"CPUCycleCount", PERF_TYPE_HARDWARE, PERF_COUNT_HW_CPU_CYCLES};
 
+    struct event leader;
+    if (mEnablePerapiPerf)
+    {
+        volatile uint64_t pmcr_el0;
+        asm volatile("mrs %0, PMCR_EL0" : "=r"(pmcr_el0));
+        pmu_counter_bits = ((pmcr_el0 & 0x80) == 0x80 ? 64 : 32);
+        if (!mConfig.isMember("type")) exit(EXIT_FAILURE);
+
+        DBG_LOG("pmu counter bits are: %u\n", pmu_counter_bits);
+        DBG_LOG("pmcr_el0 is: %lu\n", pmcr_el0);
+
+        leader = {"CPUCycleCount", mConfig.get("type", 8).asUInt(), 0x0011, false, false, pmu_counter_bits == 64 ? hw_cnt_length::b64 : hw_cnt_length::b32, false};
+    }
+    else
+    {
+        leader = {"CPUCycleCount", PERF_TYPE_HARDWARE, PERF_COUNT_HW_CPU_CYCLES};
+    }
+
+    if (mEnablePerapiPerf) mAllThread = false;
     mSet = mConfig.get("set", -1).asInt();
-    mInherit = mConfig.get("inherit", 1).asInt();
+    mInherit = mConfig.get("inherit", 0).asInt();
```

Comment:
Keep default value as 1, as most test cases, especially the ST tests, use "1" by default.